Tests/UnitTests/CMakeLists.txt

```diff
@@ -112,7 +112,7 @@ function(add_non_compile_test name src)
 endfunction()
 
 add_subdirectory(Core)
-add_subdirectory_if(Examples ACTS_BUILD_EXAMPLES)
+add_subdirectory_if(Examples ACTS_BUILD_EXAMPLES AND ACTS_BUILD_EXAMPLES_UNITTESTS)
```

Comment:
Is this related to SonarCube?

Reply:
I missed that case and noticed working on this PR. So I definitely want it.
I can factor this out into a separate PR, but it feels like a too-small change for me to warrant that.